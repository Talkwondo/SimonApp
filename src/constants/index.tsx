export const podRows = [1, 2];
export const numberOfPods = [1, 2];
export const scorePoint = 1;
export enum Colors {
  SIMON_GREEN = '#61fb7b',
  SIMON_RED = '#fb6161',
  SIMON_YELLOW = '#fbf561',
  SIMON_PINK = '#fb61d5',
  SIMON_BABYBLUE = '#61effb',
  SIMON_BLUE = '#61A9FB',
  SIMON_TRANSPERANT = 'transparent',
  SIMON_BG = '#282c34',
}
