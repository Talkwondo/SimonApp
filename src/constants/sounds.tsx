let Sound, vol1, vol2, vol3, vol4; //bug in library of react native sound ts

if (process.env.NODE_ENV !== 'test') {
  Sound = require('react-native-sound');
  vol1 = require('../sound/1.mp3');
  vol2 = require('../sound/2.mp3');
  vol3 = require('../sound/3.mp3');
  vol4 = require('../sound/4.mp3');

  const vol1Play = new Sound(vol1, (error: any) => {
    if (error) {
      console.log('failed to load the sound', error);
      return;
    }
    vol1Play.setVolume(1);
  });

  const vol2Play = new Sound(vol2, (error: string) => {
    if (error) {
      console.log('failed to load the sound', error);
      return;
    }
    vol2Play.setVolume(1);
  });

  const vol3Play = new Sound(vol3, (error: string) => {
    if (error) {
      console.log('failed to load the sound', error);
      return;
    }
    vol3Play.setVolume(1);
  });

  const vol4Play = new Sound(vol4, (error: string) => {
    if (error) {
      console.log('failed to load the sound', error);
      return;
    }
    vol4Play.setVolume(1);
  });

  module.exports = {
    vol1Play,
    vol2Play,
    vol3Play,
    vol4Play,
  };
}
