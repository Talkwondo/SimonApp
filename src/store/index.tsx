import {createSlice, configureStore, PayloadAction} from '@reduxjs/toolkit';
import {History, ScoresState} from './store.modal';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';

const initialState: ScoresState = {
  tableScore: [],
};

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

export const scoreSlice = createSlice({
  name: 'Scores',
  initialState,
  reducers: {
    addScoreToHistory: (state, action: PayloadAction<History>) => {
      state.tableScore.push(action.payload);
    },
  },
});

const persistedReducer = persistReducer(persistConfig, scoreSlice.reducer);
export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});
export const persistor = persistStore(store);
export default store;
