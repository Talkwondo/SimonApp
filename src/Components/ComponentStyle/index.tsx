import {StyleSheet, Dimensions} from 'react-native';
import {Colors} from '../../constants';
const {width} = Dimensions.get('screen');

export const ComponentStyle = StyleSheet.create({
  podsContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginVertical: 10,
  },
  podStyle: {
    maxWidth: 150,
    maxHeight: 150,
    width: width / 4,
    height: width / 4,
    borderRadius: width / 2,
    borderColor: 'white',
    borderWidth: 1,
    marginHorizontal: 10,
  },
  button: {
    width: 160,
    paddingHorizontal: 10,
    backgroundColor: Colors.SIMON_PINK,
    borderRadius: 25,
    alignSelf: 'center',
    justifyContent: 'center',
    height: 60,
    margin: 15,
  },
  title: {
    fontSize: 25,
    textAlign: 'center',
    color: 'white',
    fontWeight: '600',
  },
});
