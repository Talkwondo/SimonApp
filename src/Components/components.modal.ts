import {Dispatch, SetStateAction} from 'react';
import {Colors} from '../constants';

export interface RowPodProps {
  row: number;
  userLight: any;
  clickable: boolean;
  rowColor: number;
  colColor: number;
  setColor: Dispatch<SetStateAction<Colors>>;
  pickColor: (
    row: number,
    col: number,
    setColor: Dispatch<SetStateAction<Colors>>,
  ) => void;
  color: Colors;
}

export interface ButtonProps {
  title: string;
  cb: () => void;
  color: Colors;
}
