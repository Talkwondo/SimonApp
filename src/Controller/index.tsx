import {SetStateAction} from 'react';
import {Colors} from '../constants';
let sounds: any; //bug in library of react native sound ts
if (process.env.NODE_ENV !== 'test') {
  sounds = require('../constants/sounds');
}
export const pickColor = (
  row: number,
  col: number,
  setColor: React.Dispatch<SetStateAction<Colors>>,
) => {
  switch (true) {
    case row === 1 && col === 1:
      sounds?.vol1Play.play();
      setColor(Colors.SIMON_GREEN);
      break;
    case row === 1 && col === 2:
      sounds?.vol2Play.play();
      setColor(Colors.SIMON_YELLOW);
      break;
    case row === 2 && col === 1:
      sounds?.vol3Play.play();
      setColor(Colors.SIMON_RED);
      break;
    case row === 2 && col === 2:
      sounds?.vol4Play.play();
      setColor(Colors.SIMON_BABYBLUE);
      break;
    default:
      setColor(Colors.SIMON_TRANSPERANT);
  }
};

export const random = (podRows: number[]) =>
  podRows.map((num: number, index: number) =>
    index === 1 ? Math.round(Math.random() + 1) : Math.round(Math.random() + 1),
  );
