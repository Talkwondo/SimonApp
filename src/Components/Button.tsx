import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {ButtonProps} from './components.modal';
import {ComponentStyle} from './ComponentStyle';

export const Button = ({title, cb, color}: ButtonProps) => {
  return (
    <TouchableOpacity
      style={[ComponentStyle.button, {backgroundColor: color}]}
      onPress={cb}>
      <Text style={ComponentStyle.title}>{title}</Text>
    </TouchableOpacity>
  );
};
