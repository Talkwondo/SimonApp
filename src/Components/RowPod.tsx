import React from 'react';
import {TouchableOpacity, View} from 'react-native';
import {ComponentStyle} from './ComponentStyle';
import {numberOfPods} from '../constants';
import {RowPodProps} from './components.modal';
import uuid from 'react-native-uuid';
import {Colors} from '../constants';

const RowPod = ({
  row,
  userLight,
  clickable,
  rowColor,
  colColor,
  setColor,
  pickColor,
  color,
}: RowPodProps) => {
  return (
    <View style={ComponentStyle.podsContainer}>
      {numberOfPods.map(col => (
        <TouchableOpacity
          testID={`pod-${row}-${col}`}
          disabled={!clickable}
          onPress={() => {
            pickColor(row, col, setColor);
            userLight(row, col);
          }}
          style={[
            ComponentStyle.podStyle,
            {
              backgroundColor:
                row === rowColor && col === colColor
                  ? color
                  : Colors.SIMON_TRANSPERANT,
            },
          ]}
          key={String(uuid.v4())}
        />
      ))}
    </View>
  );
};

export default RowPod;
