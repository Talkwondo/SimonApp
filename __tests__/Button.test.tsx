//@ts-ignore
import React = require('react');
import renderer from 'react-test-renderer';
import {Button} from '../src/Components/Button';
import {Colors} from '../src/constants';
describe('Snapshot', () => {
  it('to match snapshot', () => {
    const component = renderer.create(
      <Button title={'Button'} cb={() => {}} color={Colors.SIMON_BLUE} />,
    );

    const match = component.toJSON();
    expect(match).toMatchSnapshot();
  });
});
