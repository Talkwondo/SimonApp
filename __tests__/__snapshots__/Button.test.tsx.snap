// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot to match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#61A9FB",
      "borderRadius": 25,
      "height": 60,
      "justifyContent": "center",
      "margin": 15,
      "opacity": 1,
      "paddingHorizontal": 10,
      "width": 160,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 25,
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  >
    Button
  </Text>
</View>
`;
